cmd/skaffold/app/cmd/parse_config.go

```diff
@@ -193,15 +200,17 @@ func processEachDependency(d latest.ConfigDependency, cfgOpts configOpts, opts c
 		// empty path means configs in the same file
 		path = cfgOpts.file
 	}
-	fi, err := os.Stat(path)
-	if err != nil {
-		if os.IsNotExist(errors.Unwrap(err)) {
-			return nil, fmt.Errorf("could not find skaffold config %s that is referenced as a dependency in config %s", path, cfgOpts.file)
+	if !util.IsURL(path) {
+		fi, err := os.Stat(path)
+		if err != nil {
+			if os.IsNotExist(errors.Unwrap(err)) {
+				return nil, fmt.Errorf("could not find skaffold config %s that is referenced as a dependency in config %s", path, cfgOpts.file)
+			}
+			return nil, fmt.Errorf("parsing dependencies for skaffold config %s: %w", cfgOpts.file, err)
+		}
+		if fi.IsDir() {
+			path = filepath.Join(path, "skaffold.yaml")
 		}
-		return nil, fmt.Errorf("parsing dependencies for skaffold config %s: %w", cfgOpts.file, err)
-	}
-	if fi.IsDir() {
-		path = filepath.Join(path, "skaffold.yaml")
 	}
 	cfgOpts.isDependency = cfgOpts.isDependency || path != cfgOpts.file
```

Comment:
This line is a bit puzzling and warrants a comment as I'm not sure how this situation can arise.

Reply:
added

Reply:
What's previous and current?  Oh!  Is configOpts from the config that required `d` as a dependency?  That was not clear.